packages/knex/src/adapter.ts

```diff
@@ -69,15 +69,15 @@ export class KnexAdapter<
   }
 
   db(params?: ServiceParams) {
-    const { Model, name, schema } = this.getOptions(params)
+    const { Model, name, schema, tableOptions } = this.getOptions(params)
 
     if (params && params.transaction && params.transaction.trx) {
       const { trx } = params.transaction
       // debug('ran %s with transaction %s', fullName, id)
       return schema ? (trx.withSchema(schema).table(name) as Knex.QueryBuilder) : trx(name)
     }
 
-    return schema ? (Model.withSchema(schema).table(name) as Knex.QueryBuilder) : Model(name)
+    return schema ? (Model.withSchema(schema).table(name) as Knex.QueryBuilder) : Model(name, tableOptions)
```

Comment:
Will it work if the tableOptions has not been provided, I would prefer we check it before passing the tableOptions as parameter on the knex adapter options like this:-

```
let queryBuilder: Knex.QueryBuilder;
  if (params?.transaction?.trx) {
    const { trx } = params.transaction;
    queryBuilder = schema ? trx.withSchema(schema).table(name) : trx.table(name);
  } else {
    queryBuilder = schema ? Model.withSchema(schema).table(name) : Model.table(name);
  }

  // Apply tableOptions if provided
  if (tableOptions) {
    queryBuilder = queryBuilder.withOptions(tableOptions); // Adjust this if .withOptions() isn't the exact method for tableOptions
  }
  return queryBuilder;
```